src/core/qgspallabeling.h

```diff
@@ -823,7 +826,7 @@ class CORE_EXPORT QgsPalLayerSettings
     void setFormat( const QgsTextFormat &format ) { mFormat = format; }
 
     // temporary stuff: set when layer gets prepared or labeled
-    QgsFeature *mCurFeat = nullptr;
+    const QgsFeature *mCurFeat = nullptr;
```

Comment:
Is this a pointer to a feature owned elsewhere? Or an optional feature owned by this class?

Reply:
As far as I can see it's only used to identify cached datadefined values. If the ptr is equal to the address of some feature, cached values are used.
I hope to find a better way to deal with it.

Reply:
Oh ok. I wonder if it's ever actually matching now since we just use copies of features everywhere.

Reply:
I think it might work since we only send references through the call stack (everything in this PR actually).
But I guess it's layer specific so we could use the feature id to identify the feature instead.